### PR TITLE
Withdraw job feature

### PR DESCRIPTION
**Pull request by daisycrego**

### CODE REVIEW

File: applications/static/applications/js/applications.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    });
+
+    $("#no").click(function(){
+        $("#withdraw").hide();
+        $("#withdraw_button").show();
+    });
+
+    // If the application status has the W class, indicating an application status of
+    // WITHDRAWN, add grey coloring to the job details and personal details
+    if ($(".W")){
```
[the remark this answers is not in the document]
Comment (jackxujh) [reply applications/static/applications/js/applications.js]: What about using CSS like the original ones by @surajgovardhangaikwad : https://github.com/gcivil-nyu-org/fall2019-cs-gy-6063-team-one/blob/0d9633e685c898774d1a8d00eb6503173a868c73/dashboard/static/dashboard/css/dashboard.css#L111-L123

File: dashboard/templates/dashboard/applications_list.html [hunk 2]
```diff
@@ -23,6 +24,8 @@ <h1 class="mb-4">Pending Applications</h1>
         <h1 class="mb-4">Rejected Applications</h1>
         {% elif application_type == "ac" %}
         <h1 class="mb-4">Accepted Applications</h1>
+        {% elif application_type == "w" %}
```
[the remark this answers is not in the document]
Comment (jackxujh) [reply dashboard/templates/dashboard/applications_list.html]: One of the future-proof ways I can think of is something like this: https://github.com/gcivil-nyu-org/fall2019-cs-gy-6063-team-one/blob/247dd79204c097c7ca60d72d531b1f97f943ce04/notifications/templates/notifications/notification_center.html#L30-L34

The idea is, to use less string constants when they can be referenced from the related initial definitions.

File: applications/views.py [hunk 3]
```diff
@@ -83,3 +83,22 @@ def get(self, request, *args, **kwargs):
...
+
+class WithdrawApplicationView(LoginRequiredMixin, DetailView):
+    def post(self, request, *args, **kwargs):
+        application = Application.objects.get(id=self.kwargs["pk"])
+        if application:
+            if application.status == Application.STATUS_APPLIED:
+                application.status = Application.STATUS_WITHDRAWN
+                application.save()
+
+        """return reverse(
```
[the remark this answers is not in the document]
Comment (daisycrego) [reply applications/views.py]: thanks removed that

File: applications/static/applications/js/applications.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (daisycrego) [reply applications/static/applications/js/applications.js]: i see.

File: dashboard/templates/dashboard/applications_list.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (daisycrego) [reply dashboard/templates/dashboard/applications_list.html]: good idea, done

[the remark this answers is not in the document]
Comment (jackxujh) [reply dashboard/templates/dashboard/applications_list.html]: LGTM

File: dashboard/templates/dashboard/applications_list.html [hunk 4]
```diff
@@ -18,13 +18,13 @@
                 </div>
             </div>
 
-    {% if application_type == "ap" %}
+    {% if application_type == Application.STATUS_APPLIED %}
```
Comment (jackxujh) [inline dashboard/templates/dashboard/applications_list.html]: @surajgovardhangaikwad I was referring to this about the scattered string constants.

**Pull request merged**